internal/peer/node/start.go

```diff
@@ -802,8 +803,9 @@ func serve(args []string) error {
 		coreConfig.ValidatorPoolSize,
 	)
 
+	var discoveryService gateway.DiscoveryService
 	if coreConfig.DiscoveryEnabled {
-		registerDiscoveryService(
+		discoveryService = registerDiscoveryService(
```

Comment:
I'm also not a fan keeping the name `registerXXX` when we're relying on a side-effect. This should be broken into a function that creates the service and then one that registers it.